WcaOnRails/app/controllers/delegate_reports_controller.rb

```diff
@@ -29,13 +29,14 @@ def update
 
     @delegate_report = @competition.delegate_report
     @delegate_report.current_user = current_user
-    @delegate_report.discussion_url = "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(@delegate_report.competition.name)
     was_posted = @delegate_report.posted?
     if @delegate_report.update_attributes(delegate_report_params)
       flash[:success] = "Updated report"
       if @delegate_report.posted? && !was_posted
         # Don't email when posting old delegate reports.
         # See https://github.com/cubing/worldcubeassociation.org/issues/704 for details.
+        # Set the discussion_url column only at posting
+        @delegate_report.discussion_url = "https://groups.google.com/forum/#!topicsearchin/wca-delegates/" + URI.encode(@delegate_report.competition.name)
```

Comment:
I would suggest to move this logic to the model by wrapping this in a `before_create` callback.